Sources/Swift/Protocol/Metric.swift

```diff
@@ -0,0 +1,289 @@
+        self.unit = unit
+        self.attributes = attributes
+        
+        // Perform type conversion when needed
+        switch (type, value) {
+        case (.counter, .integer), (.gauge, .double), (.distribution, .double):
+            self._value = value
+        case (.counter, .double(let doubleValue)):
+            // Convert double to integer by flooring
+            SentrySDKLog.warning("Counter metric created with double value \(doubleValue). Converting to integer by flooring: \(Int64(floor(doubleValue))).")
+            self._value = .integer(Int64(floor(doubleValue)))
+        case (.gauge, .integer(let intValue)):
+            // Convert integer to double
+            SentrySDKLog.warning("Gauge metric created with integer value \(intValue). Converting to double: \(Double(intValue)).")
+            self._value = .double(Double(intValue))
+        case (.distribution, .integer(let intValue)):
+            // Convert integer to double
+            SentrySDKLog.warning("Distribution metric created with integer value \(intValue). Converting to double: \(Double(intValue)).")
+            self._value = .double(Double(intValue))
+        }
+    }
+    
+    /// Adds or updates an attribute in the metric entry.
+    /// - Parameters:
+    ///   - attribute: The attribute value to add
+    ///   - key: The key for the attribute
+    public mutating func setAttribute(_ attribute: SentryAttribute?, forKey key: String) {
+        if let attribute = attribute {
+            attributes[key] = attribute
+        } else {
+            attributes.removeValue(forKey: key)
+        }
+    }
+}
+
+extension Metric: Encodable {
+    private enum CodingKeys: String, CodingKey {
+        case timestamp
+        case traceId = "trace_id"
+        case name
+        case value
+        case type
+        case unit
+        case attributes
+    }
+    
+    public func encode(to encoder: any Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        
+        try container.encode(timestamp, forKey: .timestamp)
+        try container.encode(traceId.sentryIdString, forKey: .traceId)
+        try container.encode(name, forKey: .name)
+        try container.encode(value, forKey: .value)
+        
+        try container.encode(metricType, forKey: .type)
+        try container.encodeIfPresent(unit, forKey: .unit)
+        try container.encode(attributes, forKey: .attributes)
+    }
+}
+
+extension Metric: BatcherItem {}
+
+// MARK: - MetricType
+
+/// The type of metric being recorded.
+///
+/// Different metric types serve different purposes:
+/// - **Counter**: Incrementing integer values that only increase (e.g., total requests, errors)
+/// - **Gauge**: Current value at a point in time that can go up or down (e.g., active connections, queue size)
+/// - **Distribution**: Statistical distribution of values for aggregation (e.g., response times, payload sizes)
+public enum MetricType: Encodable {
```

Comment:
`m`: This actually implements both decode and encode
```suggestion
public enum MetricType: Codeable {
```